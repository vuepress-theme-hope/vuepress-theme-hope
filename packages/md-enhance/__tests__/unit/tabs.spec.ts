import MarkdownIt from "markdown-it";
import { describe, expect, it } from "vitest";

import { tabs } from "../../src/node/markdown-it/index.js";

const markdownIt = MarkdownIt({ linkify: true }).use(tabs);

describe("tabs", () => {
  it("Should render single block", () => {
    expect(
      markdownIt.render(`
::: tabs

@tab js

\`\`\`js
const a = 1;
\`\`\`

:::
    `)
    ).toMatchSnapshot();

    expect(
      markdownIt.render(`
::: tabs
@tab js
\`\`\`js
const a = 1;
\`\`\`
:::
    `)
    ).toMatchSnapshot();
  });

  it("Should render multiple block", () => {
    expect(
      markdownIt.render(`
::: tabs

@tab js

\`\`\`js
const a = 1;
\`\`\`

@tab ts

\`\`\`ts
const a = 1;
\`\`\`

:::
`)
    ).toMatchSnapshot();

    expect(
      markdownIt.render(`
::: tabs
@tab js
\`\`\`js
const a = 1;
\`\`\`
@tab ts
\`\`\`ts
const a = 1;
\`\`\`
:::
`)
    ).toMatchSnapshot();
  });

  it("Should support tabs id", () => {
    expect(
      markdownIt.render(`
::: tabs#event

@tab js

\`\`\`js
const a = 1;
\`\`\`

:::
    `)
    ).toMatchSnapshot();

    expect(
      markdownIt.render(`
::: tabs#event-id
@tab js
\`\`\`js
const a = 1;
\`\`\`
:::
    `)
    ).toMatchSnapshot();

    expect(
      markdownIt.render(`
::: tabs#id with space
@tab js
\`\`\`js
const a = 1;
\`\`\`
:::
    `)
    ).toMatchSnapshot();

    expect(
      markdownIt.render(`
::: tabs # id starts and having space in the end
@tab js
\`\`\`js
const a = 1;
\`\`\`
:::
    `)
    ).toMatchSnapshot();
  });

  it("Should support active", () => {
    expect(
      markdownIt.render(`
::: tabs

@tab:active js

\`\`\`js
const a = 1;
\`\`\`

:::
    `)
    ).toMatchSnapshot();

    expect(
      markdownIt.render(`
::: tabs
@tab:active js
\`\`\`js
const a = 1;
\`\`\`
:::
    `)
    ).toMatchSnapshot();

    expect(
      markdownIt.render(`
::: tabs

@tab js

\`\`\`js
const a = 1;
\`\`\`

@tab:active ts

\`\`\`ts
const a = 1;
\`\`\`

:::
    `)
    ).toMatchSnapshot();

    expect(
      markdownIt.render(`
::: tabs
@tab js
\`\`\`js
const a = 1;
\`\`\`
@tab:active ts
\`\`\`ts
const a = 1;
\`\`\`
:::
    `)
    ).toMatchSnapshot();
  });

  it("should ignore other items", () => {
    expect(
      markdownIt.render(`
::: tabs

\`\`\`coffee
const a = 1;
\`\`\`

@tab:active js

\`\`\`js
const a = 1;
\`\`\`

\`\`\`ts
const a = 1;
\`\`\`

:::
    `)
    ).toMatchSnapshot();

    expect(
      markdownIt.render(`
::: tabs
\`\`\`coffee
const a = 1;
\`\`\`
@tab:active js
\`\`\`js
const a = 1;
\`\`\`
\`\`\`ts
const a = 1;
\`\`\`
:::
    `)
    ).toMatchSnapshot();
  });
});
