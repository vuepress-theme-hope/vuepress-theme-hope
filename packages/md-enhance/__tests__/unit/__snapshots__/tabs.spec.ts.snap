// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`tabs > Should render multiple block 1`] = `
"<Tabs id=\\"0\\" :data='[{\\"title\\":\\"js\\"},{\\"title\\":\\"ts\\"}]'>
<template #tab0=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-js\\">const a = 1;
</code></pre>
</template>
<template #tab1=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-ts\\">const a = 1;
</code></pre>
</template>
</Tabs>
"
`;

exports[`tabs > Should render multiple block 2`] = `
"<Tabs id=\\"0\\" :data='[{\\"title\\":\\"js\\"},{\\"title\\":\\"ts\\"}]'>
<template #tab0=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-js\\">const a = 1;
</code></pre>
</template>
<template #tab1=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-ts\\">const a = 1;
</code></pre>
</template>
</Tabs>
"
`;

exports[`tabs > Should render single block 1`] = `
"<Tabs id=\\"0\\" :data='[{\\"title\\":\\"js\\"}]'>
<template #tab0=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-js\\">const a = 1;
</code></pre>
</template>
</Tabs>
"
`;

exports[`tabs > Should render single block 2`] = `
"<Tabs id=\\"0\\" :data='[{\\"title\\":\\"js\\"}]'>
<template #tab0=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-js\\">const a = 1;
</code></pre>
</template>
</Tabs>
"
`;

exports[`tabs > Should support active 1`] = `
"<Tabs id=\\"0\\" :data='[{\\"title\\":\\"js\\"}]' :active=\\"0\\">
<template #tab0=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-js\\">const a = 1;
</code></pre>
</template>
</Tabs>
"
`;

exports[`tabs > Should support active 2`] = `
"<Tabs id=\\"0\\" :data='[{\\"title\\":\\"js\\"}]' :active=\\"0\\">
<template #tab0=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-js\\">const a = 1;
</code></pre>
</template>
</Tabs>
"
`;

exports[`tabs > Should support active 3`] = `
"<Tabs id=\\"0\\" :data='[{\\"title\\":\\"js\\"},{\\"title\\":\\"ts\\"}]' :active=\\"1\\">
<template #tab0=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-js\\">const a = 1;
</code></pre>
</template>
<template #tab1=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-ts\\">const a = 1;
</code></pre>
</template>
</Tabs>
"
`;

exports[`tabs > Should support active 4`] = `
"<Tabs id=\\"0\\" :data='[{\\"title\\":\\"js\\"},{\\"title\\":\\"ts\\"}]' :active=\\"1\\">
<template #tab0=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-js\\">const a = 1;
</code></pre>
</template>
<template #tab1=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-ts\\">const a = 1;
</code></pre>
</template>
</Tabs>
"
`;

exports[`tabs > Should support tabs id 1`] = `
"<Tabs id=\\"0\\" :data='[{\\"title\\":\\"js\\"}]' tab-id=\\"event\\">
<template #tab0=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-js\\">const a = 1;
</code></pre>
</template>
</Tabs>
"
`;

exports[`tabs > Should support tabs id 2`] = `
"<Tabs id=\\"0\\" :data='[{\\"title\\":\\"js\\"}]' tab-id=\\"event-id\\">
<template #tab0=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-js\\">const a = 1;
</code></pre>
</template>
</Tabs>
"
`;

exports[`tabs > Should support tabs id 3`] = `
"<Tabs id=\\"0\\" :data='[{\\"title\\":\\"js\\"}]' tab-id=\\"id with space\\">
<template #tab0=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-js\\">const a = 1;
</code></pre>
</template>
</Tabs>
"
`;

exports[`tabs > Should support tabs id 4`] = `
"<Tabs id=\\"0\\" :data='[{\\"title\\":\\"js\\"}]' tab-id=\\"id starts and having space in the end\\">
<template #tab0=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-js\\">const a = 1;
</code></pre>
</template>
</Tabs>
"
`;

exports[`tabs > should ignore other items 1`] = `
"<Tabs id=\\"0\\" :data='[{\\"title\\":\\"js\\"}]' :active=\\"0\\">
<template #tab0=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-js\\">const a = 1;
</code></pre>
<pre><code class=\\"language-ts\\">const a = 1;
</code></pre>
</template>
</Tabs>
"
`;

exports[`tabs > should ignore other items 2`] = `
"<Tabs id=\\"0\\" :data='[{\\"title\\":\\"js\\"}]' :active=\\"0\\">
<template #tab0=\\"{ title, value, isActive }\\">
<pre><code class=\\"language-js\\">const a = 1;
</code></pre>
<pre><code class=\\"language-ts\\">const a = 1;
</code></pre>
</template>
</Tabs>
"
`;
